### PR TITLE
fix: ClusterRole Aggregation

### PR DESCRIPTION
**Pull request by sempex**
Kubernetes Supports ClusterRole Aggregation, enabling the aggregation of one ClusterRole into another. For more details, refer to the official Kubernetes Documentation [here](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#aggregated-clusterroles).

This functionality allows a ClusterRole lacking a ClusterRoleBinding or a RoleBinding to remain usable. Currently, Kor does not handle this scenario. Consequently, the ClusterRoles in use are not properly listed through Kor. The proposed Pull Request addresses this issue.

### CODE REVIEW

**Comment by yonahd** [comment]
Hi @sempex
Thank for the pr. Can you please add a test for this?

**Comment by sempex** [comment]
> Hi @sempex Thank for the pr. Can you please add a test for this?

Sure will try to add one 👍

**Comment by sempex** [comment]
@yonahd I added my Changed to the Testcases

File: pkg/kor/create_test_resources.go [hunk 1]
```diff
@@ -16,8 +16,12 @@ import (
 var testNamespace = "test-namespace"
 
 var AppLabels = map[string]string{}
+var AggregatedLabels = map[string]string{"rbac.authorization.k8s.io/aggregate-to-test-clusterRole1": "true"}
 var UsedLabels = map[string]string{"kor/used": "true"}
 var UnusedLabels = map[string]string{"kor/used": "false"}
+var matchLabels = v1.LabelSelector{
```
Comment (yonahd) [inline pkg/kor/create_test_resources.go:22]: This and line 19 should be in the clusterroles_test file

File: pkg/kor/clusterroles.go [hunk 2]
```diff
@@ -58,10 +61,51 @@ func retrieveUsedClusterRoles(clientset kubernetes.Interface, filterOpts *filter
...
+	aggregatedLabels := make([]string, 0)
 
+	for clusterRole := range usedClusterRoles {
+		clusterRoleManifest := clusterRolesMap[clusterRole]
+		if clusterRolesMap[clusterRole].AggregationRule == nil {
+			continue
+		}
+		for _, label := range clusterRoleManifest.AggregationRule.ClusterRoleSelectors {
+			for key, value := range label.MatchLabels {
+				aggregatedLabels = append(aggregatedLabels, key+": "+value)
```
Comment (yonahd) [inline pkg/kor/clusterroles.go:86]: nit: using` fmt.Sprintf("%s: %s", key, value)` may be a little cleaner

**Review by yonahd — COMMENTED** [review]
Looks good!
A couple small comments

**Comment by sempex** [comment]
> Looks good! A couple small comments

Fixed it 🚀

**Review by yonahd — APPROVED** [review]
LGTM
Thank you!

**Pull request merged**